cpp/src/umap/knn_graph/algo.cuh

```diff
@@ -139,9 +129,10 @@ inline void launcher(const raft::handle_t& handle,
         target[i * n_neighbors + j] = source[i * graph_degree + j];
       }
     }
-    raft::copy(handle,
-               raft::make_device_matrix_view(out.knn_indices, inputsA.n, n_neighbors),
-               temp_indices_h.view());
+    raft::copy(out.knn_indices,
+               temp_indices_h.data_handle(),
+               inputsA.n * n_neighbors,
+               raft::resource::get_cuda_stream(handle));
```

Comment:
Why is this in the diff?

Reply:
It seems like using the [view-copy API](https://github.com/rapidsai/raft/blob/2a402271a6ee60d7e4e31c6d5ce7130f4f509fa5/cpp/include/raft/matrix/copy.cuh#L69-L71), we copy from in -> out.
Here we want to copy from `temp_indices_h` -> `out.knn_indices`, but it seems like the previous code was doing it the other way round? Plus they both take device matrix views and this is a host to device copy, so I'm not even sure how the previous one was working..!?

Reply:
It was using [this API](https://github.com/rapidsai/raft/blob/2a402271a6ee60d7e4e31c6d5ce7130f4f509fa5/cpp/include/raft/core/copy.cuh#L56-L62). I think we can keep this code as it was before.